docs/docsite/_ext/ansible_redirects.py

```diff
@@ -0,0 +1,27 @@
+"""Sphinx extension for generating redirects in Ansible package docs."""
+
+import os
+from pathlib import Path
+
+import yaml
+from sphinx.application import Sphinx
+
+DOCSITE_ROOT_DIR = Path(__file__).parents[1].resolve()
+ANSIBLE_REDIRECTS_CONFIG_PATH = DOCSITE_ROOT_DIR / "ansible_redirects.yaml"
+REDIRECT_KEYS = yaml.safe_load(
+    ANSIBLE_REDIRECTS_CONFIG_PATH.read_text(encoding="utf-8")
+)
+ENABLE_REDIRECTS = os.environ.get("ENABLE_REDIRECTS", "false").lower() in {"1", "true"}
+
+
+def setup(app: Sphinx):
+    app.setup_extension("sphinx_reredirects")
```

Comment:
Why does this have to be unconditional? If anything, I'd put it into an if-block.

Also, modifying the config that seems to be related to this extension after initializing it seems problematic.

Reply:
Doesn't this just achieve the same thing as loading the `redirects` configuration option? https://documatt.com/sphinx-reredirects/usage.html#usage